homeassistant/components/homekit/accessories.py

```diff
@@ -415,16 +416,20 @@ def __init__(
             CHAR_STATUS_LOW_BATTERY, value=0
         )
 
+    def _update_available_from_state(self, new_state: State | None) -> None:
+        """Update the available property based on the state."""
+        self._available = new_state is not None and new_state.state != STATE_UNAVAILABLE
+
     @property
     def available(self) -> bool:
         """Return if accessory is available."""
-        state = self.hass.states.get(self.entity_id)
-        return state is not None and state.state != STATE_UNAVAILABLE
```

Comment:
The goal is to get rid of the state machine lookup here since we can already remember it since we are tracking it